rllib/core/learner/learner.py

```diff
@@ -1143,19 +1143,22 @@ def update(
             batch_iter = MiniBatchDummyIterator
 
         results = []
-        for minibatch in batch_iter(batch, minibatch_size, num_iters):
-            # Convert minibatch into a tensor batch (NestedDict).
-            tensor_minibatch = self._convert_batch_type(minibatch)
+        # Convert minibatch into a tensor batch (NestedDict) on the correct device
+        # (e.g. GPU). We mov the batch to device here to avoid moving ever minibatch
+        # that is created in the `batch_iter` below.
```

Comment:
```suggestion
        # (e.g. GPU). We move the batch already here to avoid having to move every
        # single minibatch that is created in the `batch_iter` below.
```